src/pages/api/auth/[...nextauth].ts

```diff
@@ -44,60 +41,55 @@ export const authOptions: NextAuthOptions = {
     },
     async jwt({ isNewUser, token, account, ...response }) {
       const profile = response.profile as GhExtendedProfile | undefined;
-      const createAndSetNewUser = async ({
-        username,
-        email,
-        permissions,
-      }: CreateUserProp) => {
-        const res = await createNewUser({ username, email, permissions });
-        if (res.data) {
+
+      if (isNewUser && profile?.login) {
+        const { email, login } = profile;
+        try {
+          const signedUpUser = await signUpNewUser({
```

Comment:
this function should receive the GitHub token

Reply:
yup yup, I'll address that